statusHandler/p2pQuota/p2pQuotaProcessor.go

```diff
@@ -0,0 +1,135 @@
+package p2pQuota
+
+import (
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/statusHandler"
+)
+
+type quota struct {
+	numReceivedMessages   uint32
+	sizeReceivedMessages  uint64
+	numProcessedMessages  uint32
+	sizeProcessedMessages uint64
+}
+
+// p2pQuotaProcessor implements process.QuotaStatusHandler and is able to periodically sends to a
+// statusHandler the processed p2p quota information
+type p2pQuotaProcessor struct {
+	mutStatistics   sync.Mutex
+	statistics      map[string]*quota
```

Comment:
Why use string and not just a simple int for the statistic key ? You're hiding the string behind a const anyway, so the code would look the same where the const is used (like status.GetUInt64(MaxBlabla)) but the map won't have to hash the string for the key (so it will be faster). Even faster: why a hash and not a simple array with the const MaxBlablaEtc indicating the index in the array ?